src/templates/ReportForm.tsx

```diff
@@ -77,123 +76,170 @@ const ReportForm = ({ data, location }) => {
+    }
+
+    // If nothing changed, do not submit
+    if (
+      !requestData.attributedTo &&
+      !requestData.title &&
+      !requestData.body &&
+      !requestData.risk
+    ) {
+      throw "There were no changes to save.";
+    }
+    return requestData;
+  };
+
+  const initializeEditState = (finding) => {
+    // normalize whitespace
+    const normalizedBody = finding.body.replaceAll("\r\n", "\n");
+    const { links, body } = separateLinksFromBody(normalizedBody, finding.risk);
+    setMode(FormMode.Edit);
+    setIssueId(parseInt(finding.issueNumber));
+    setFindingId(`${contestid}-${finding.issueNumber}`);
+    setState({
+      title: finding.title,
+      risk: finding.risk,
+      details: body,
+      qaGasDetails: normalizedBody,
+      linksToCode: links,
+    });
+    setAttributedTo(finding.handle);
+    setFindingId(`${contestid}-${finding.issueNumber}`);
+  };
+
+  const initializeCreateState = () => {
+    setIssueId(0);
+    setFindingId(contestid);
+    setState(initialState);
+    setMode(FormMode.Create);
+    setAttributedTo(currentUser.username);
+  };
+
+  function findDiff(str1, str2) {
+    let diff = "";
+    str2.split("").forEach(function (val, i) {
+      if (val != str1.charAt(i)) diff += val;
+    });
+    return diff;
+  }
+
   useEffect(() => {
     (async () => {
       if (currentUser.isLoggedIn) {
         const user = await Moralis.User.current();
 
         if (location.state && location.state.finding) {
           const finding = location.state.finding;
-          setFindingId(finding.issueNumber);
-          setState({
-            title: finding.title,
-            risk: finding.risk,
-            details: finding.body,
-            qaGasDetails: finding.body,
-            linksToCode: finding.linksToCode,
-          });
-          setIsLoading(false);
-          // ? will this swap the endpoint on normal submit?
-          setEndpoint("manage-findings");
-          setFindingId(`${contestid}-${finding.issueNumber}`);
+          initializeEditState(finding);
         } else if (location.search) {
+          const params = new URLSearchParams(location.search);
+          const id = parseInt(params.get("issue") as string);
+          if (!id) {
+            initializeCreateState();
+            return;
+          }
           try {
-            // @todo: improve this
-            const params = new URLSearchParams(location.search);
-            const id = params.get("issue") as string;
-            // if id?
-            // // placeholder function for getting issue by id;
-            // // will need to reshape the response to be ReportState
-            // // before passing into setting initial state
-            setEndpoint("manage-findings");
-            setIssueId(id);
-            setFindingId(`${contestid}-${id}`);
-
             const q = new URLSearchParams({
-              "contest": contestid,
-              "issue": id,
+              contest: contestid as string,
```

Comment:
Below uses toString -- is there a difference?

Reply:
Good catch. There was a lot of messy type-casting going on around there.